src/pixelflow/core.py

```diff
@@ -102,10 +110,49 @@ def pixelflow(
         * Image padding to take care of edges
         * Supplying different image channels to different functions
     """
-    features = regionprops_table(
-        label(mask),
-        image,
-        properties=features,
-        extra_properties=custom,
-    )
-    return PixelflowResult(features=pd.DataFrame(features))
+
+    if dim_labels == "":
+        if mask.ndim == 2:
+            dim_labels = "YX"
+            print("YX image detected")
+        elif mask.ndim == 3:
+            dim_labels = "ZYX"
+            print("ZYX image detected")
+        else:
+            raise ValueError("Image must be YX or ZYX, check mask.ndim")
+
+    # check dim_labels matches mask.ndim
+    if not len(dim_labels) == mask.ndim:
```

Comment:
In general, I prefer a more explicit syntax here, as I find it easier to read:

```python
if len(dim_labels) != mask.ndim:
```

Reply:
Done